src/workers/parser/utils/contains.js

```diff
@@ -4,11 +4,21 @@ import type { AstLocation, AstPosition } from "../types";
 import type { Node } from "babel-traverse";
 
 export function containsPosition(a: AstLocation, b: AstPosition) {
-  const startsBefore =
-    a.start.line < b.line ||
-    (a.start.line === b.line && a.start.column <= b.column);
-  const endsAfter =
-    a.end.line > b.line || (a.end.line === b.line && a.end.column >= b.column);
+  const startsBefore = a.start.line < b.line;
+  if (!startsBefore && a.start.line === b.line) {
+    // If the start line is equal to the target line
+    // and there is no column criterion, we judge it as true.
+    // Otherwise we have to check if the start column is really before the target colimn.
+    startsBefore =
+      a.start.column >= 0 && b.column >= 0 ? a.start.column <= b.column : true;
+  }
+
+  const endsAfter = a.end.line > b.line;
+  if (!endsAfter && a.end.line === b.line) {
+    // We do the same check as the startsBefore case for the endsAfter case.
+    endsAfter =
+      a.end.column >= 0 && b.column >= 0 ? a.end.column >= b.column : true;
+  }
```

Comment:
@jasonLaster Thanks for the gist hint.

Before proceeding to the test, would like to check with you if this solution is the right direction since this touches the util function. Below is my analysis, please have a look, thank you:

While getting out-of-scope locations, it uses the parsed source locations and the bp position to filter out those not contained within scope [1]
The bp position only has the column criterion [2] and the source locations have one location [3].
During filtering, the `containsPosition` doesn’t consider the case without the column criterion. 
As a result [4]
```js
// This evaluates to false although a.start.line === b.line === 46,
// because a.start.column <= b.column is false (b.column === undefined)
startsBefore = (a.start.line === b.line && a.start.column <= b.column); 
```
then the line 46 where the bp exists is judged as out of scope.

So the solution is to take the case without column criterion into consideration.

[1] https://github.com/devtools-html/debugger.html/blob/c5177581a8af4b797f9aa56f82214216de0176c6/src/workers/parser/getOutOfScopeLocations.js#L86

[2] Object { sourceId: "server1.conn26.child2/source32", line: 46 }

[3] {
  end: Object { line: 48, column: 3 }
  start: Object { line: 46, column: 30 }
}

[4] https://github.com/devtools-html/debugger.html/blob/c5177581a8af4b797f9aa56f82214216de0176c6/src/workers/parser/utils/contains.js#L9